### DISCUSSION TITLE
Print Debug logs in XLA

### DISCUSSION THREAD

**Discussion by luweizheng** [jax-ml/jax#12117] [Q&A]
Hi all, I'm trying to customize XLA source code. I find many VLOG in the source code and I want to use it for debug purpose. Here is what I do: `TF_CPP_MAX_VLOG_LEVEL=3 python jax.py`. But I do not get any log. Anyone have some ideas? While `TF_CPP_MIN_LOG_LEVEL=0` and `LOG()` seem work.

**Comment by hawkinsp** [comment]
Try:

`TF_CPP_VMODULE=module=level`

e.g.,
`TF_CPP_VMODULE=jax_jit=10`

**Reply by luweizheng** [reply]
Hi Peter, Thanks for your reply. I tried with `TF_CPP_VMODULE=jax_jit=10 python jax.py`. But I cannot get what I want.
I also tried `TF_CPP_MAX_VLOG_LEVEL` with tensorflow 2.8 which is installed by pip. tensorflow can print all the VLOG.

**Reply by alexdrouillard** [reply]
Did you ever figure this out? I've been testing a custom PJRT client and can't seem to get logs printing either. Works when I run XLA unit tests, but not in my PJRT client .so that jax picks up as a plugin.

**Reply by luweizheng** [reply]
It's been two years now, I remember using some environment variables, and XLA would print more DEBUG logs.

**Comment by Rohanjames1997** [comment]
Does anyone what other environment variables control the DEBUG logs?

I tried the above suggestions but they did not help.

I'm particularly interested in printing out the emitter level traces of matmul and other ops.
For instance, I'd like to trace which of these runtimes the op is emitted to - ACL or Eigen : https://github.com/openxla/xla/blob/0a4d157ba44b378221a915965036c0aff0120a70/xla/service/cpu/dot_op_emitter.cc#L785-L786

**Comment by alexdrouillard** [comment]
I remember having some success trying TF_CPP_MIN_LOG_LEVEL. I spent far too long trying to get robust XLA logs in jax. I'm working on a seperate backend for xla currently and unfortunately what I've been doing to get logs is just hacking the VLOG definition in the xla repo to cout. You could try forking it and building it yourself with that hack if you're really desperate.

**Reply by Rohanjames1997** [reply]
Thanks @alexdrouillard ! 
For my usecase, `TF_CPP_MAX_VLOG_LEVEL=2 TF_CPP_MIN_LOG_LEVEL=0` did the job.
